src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -242,54 +248,63 @@ private static BasicStudyInfos toBasicInfos(BasicStudyEntity entity) {
     }
 
-    private Mono<BasicStudyEntity> getStudyCreationRequest(String studyName, String userId) {
-        return studyCreationRequestRepository.findStudy(userId, studyName);
+    private Optional<StudyCreationRequestEntity> getStudyCreationRequest(String studyName, String userId) {
+        return studyCreationRequestRepository.findByUserIdAndStudyName(userId, studyName);
     }
 
     @Synchronized
     public Mono<Void> deleteStudyIfNotCreationInProgress(String studyName, String userId) {
-        return getStudyCreationRequest(studyName, userId) // if creation in progress delete only the creation request
-                .switchIfEmpty(removeStudy(studyName, userId).cast(BasicStudyEntity.class))
-                .then()
-                .doFinally(r -> deleteStudyCreationRequest(studyName, userId));
+        return Mono.fromRunnable(() -> {
+            if (getStudyCreationRequest(studyName, userId).isEmpty()) {
+                removeStudy(studyName, userId);
+            }
+            deleteStudyCreationRequest(studyName, userId);
+        });
     }
 
     private Mono<StudyEntity> insertStudy(String studyName, String userId, boolean isPrivate, UUID networkUuid, String networkId,
-                                         String description, String caseFormat, UUID caseUuid, boolean casePrivate, LoadFlowStatus loadFlowStatus,
-                                         LoadFlowResultEntity loadFlowResult, LoadFlowParametersEntity loadFlowParameters, UUID securityAnalysisUuid) {
-        return studyRepository.insertStudy(studyName, userId, isPrivate, networkUuid, networkId, description, caseFormat, caseUuid, casePrivate, loadFlowStatus, loadFlowResult,
-                                           loadFlowParameters, securityAnalysisUuid)
-                .doOnSuccess(s -> emitStudyChanged(studyName, StudyService.UPDATE_TYPE_STUDIES));
-    }
-
-    private Mono<Void> removeStudy(String studyName, String userId) {
-        return studyRepository.deleteStudy(userId, studyName)
-                .doOnSuccess(s -> emitStudyChanged(studyName, StudyService.UPDATE_TYPE_STUDIES));
+                                          String description, String caseFormat, UUID caseUuid, boolean casePrivate, LoadFlowStatus loadFlowStatus,
+                                          LoadFlowResultEntity loadFlowResult, LoadFlowParametersEntity loadFlowParameters, UUID securityAnalysisUuid) {
+        StudyEntity studyEntity = new StudyEntity(userId, studyName, ZonedDateTime.now(ZoneId.of("UTC")), networkUuid, networkId, description, caseFormat, caseUuid, casePrivate, isPrivate, loadFlowStatus, loadFlowResult, loadFlowParameters, securityAnalysisUuid);
+        studyRepository.save(studyEntity);
+        emitStudyChanged(studyName, StudyService.UPDATE_TYPE_STUDIES);
+        return Mono.empty();
+    }
+
+    private void removeStudy(String studyName, String userId) {
```

Comment:
It's weird that some methods are changed to become imperative but some are not. Either change them all or none of them